sql/src/main/java/io/crate/analyze/CreateSnapshotStatementAnalyzer.java

```diff
@@ -0,0 +1,205 @@
+            for (Map.Entry<String, SettingsApplier> entry : SETTINGS.entrySet()) {
+                builder.put(entry.getValue().getDefault());
+            }
+            // apply given config
+            for (Map.Entry<String, Expression> entry : node.properties().get().properties().entrySet()) {
+                SettingsApplier settingsApplier = SETTINGS.get(entry.getKey());
+                settingsApplier.apply(builder, analysis.parameterContext().parameters(), entry.getValue());
+            }
+            settings = builder.build();
+        }
+
+        boolean ignoreUnavailable = settings.getAsBoolean(IGNORE_UNAVAILABLE.name(), IGNORE_UNAVAILABLE.defaultValue());
+
+        // iterate tables
+        if (node.tableList().isPresent()) {
+            List<Table> tableList = node.tableList().get();
+            Set<String> snapshotIndices = new HashSet<>(tableList.size());
+            boolean hasPartitions = false;
+            for (Table table : tableList) {
+                TableInfo tableInfo;
+                try {
+                    tableInfo = schemas.getTableInfo(TableIdent.of(table, analysis.parameterContext().defaultSchema()));
+                } catch (SchemaUnknownException|TableUnknownException e) {
+                    if (ignoreUnavailable) {
+                        LOGGER.info("ignoring: {}", e.getMessage());
+                        continue;
+                    } else {
+                        throw e;
+                    }
+                }
+                if (!(tableInfo instanceof DocTableInfo)) {
+                    throw new IllegalArgumentException(
+                            String.format(Locale.ENGLISH, "Cannot create snapshot of tables in schema '%s'", tableInfo.ident().schema()));
+                }
+                DocTableInfo docTableInfo = (DocTableInfo)tableInfo;
+                if (table.partitionProperties().isEmpty()) {
+                    if (docTableInfo.isPartitioned()) {
+                        hasPartitions = true;
+                    }
+                    snapshotIndices.addAll(Arrays.asList(docTableInfo.concreteIndices()));
+                } else {
+                    PartitionName partitionName = PartitionPropertiesAnalyzer.toPartitionName(
+                            docTableInfo,
+                            table.partitionProperties(),
+                            analysis.parameterContext().parameters()
+                    );
+                    if (!docTableInfo.partitions().contains(partitionName)) {
+                        if (!ignoreUnavailable) {
+                            throw new PartitionUnknownException(tableInfo.ident().fqn(), partitionName.ident());
+                        } else {
+                            LOGGER.info("ignoring unknown partition of table '{}' with ident '{}'", partitionName.tableIdent(), partitionName.ident());
+                        }
+                    } else {
+                        hasPartitions = true;
```

Comment:
if we snapshot partitions only, we must not snapshot the globalState (no templates are needed to be stored) and so hasPartitions should not be set to true